src/OrchardCore/OrchardCore.Mvc.Core/ShellFileVersionProvider.cs

```diff
@@ -127,6 +147,16 @@ public string AddFileVersionToPath(PathString requestPathBase, string path)
                 }
             }
 
+            foreach (var fileStoreVersionProvider in _fileStoreVersionProviders)
+            {
+                // This will set cache if file is found.
+                var versionedPath = fileStoreVersionProvider.AddFileVersionToPathAsync(cacheKey, path).GetAwaiter().GetResult();
+                if (versionedPath != null)
```

Comment:
this `.GetAwaiter().GetResult()` annoys me. I opened an issue on aspnetcore to ask if the `IFileVersionProvider` could be made async. It's been scheduled for core 3.1 so maybe. Could remove this blob cache busting from this pr and look at seperately if you like.

Reply:
I didn't follow closely enough this ShellFileVersionProvider thing, but there is so much code in it, and so many implementation details leaking that it worries me a little.

Reply:
Ok, I haven't looked at it again yet. 

It's just cache busting, but to implement with support for cdn and blob has made it messy. I might just pull from this pr, knowing that cache busting support won't work with blob and / or cdn, and we can look at in another PR if I can get it a bit cleaner

Reply:
I'd prefer that.

Reply:
It doesn't hurt me as long as it is still less than twice the original mvc `DefaultFileVersionProvider` ;)

Maybe unless the non async call. But i think that the underlying `GetFileInfoAsync()` would need to be only used at another level just to see if a blob file needs to be loaded / reloaded to update the local cache. Then here just append the version as for a regular file or append if the local file doesn't exist yet.

Reply:
> It doesn't hurt me as long as it is still less than twice the original mvc `DefaultFileVersionProvider` ;)

Haha, yes, we still manage to stay under twice the size or the original!

> Maybe unless the non async call. But i think that the underlying `GetFileInfoAsync()` would need to be only used at another level just to see if a blob file needs to be loaded / reloaded to update the local cache. Then here just append the version as for a regular file or nothing if the local file doesn't exist yet.

I think you are right here, append version from cached file when present, and handle cached file invalidation at a different level (maybe with metadata hashes maybe with a totally different technique, will look into how nginx and/or varnish manage these things when I have time)

So I will remove this `GetFileInfoAsync()` from this PR, and the CDN cache busting support, knowing that CDN support will work, and can go in another PR, just to keep this PR focused on blob.